Include/object.h

```diff
@@ -78,12 +78,78 @@ whose size is determined when the object is allocated.
 /* PyObject_HEAD defines the initial segment of every PyObject. */
 #define PyObject_HEAD                   PyObject ob_base;
 
+/*
+Immortalization:
+
+The following indicates the immortalization strategy depending on the amount
+of available bits in the reference count field. All strategies are backwards
+compatible but the specific reference count value or immortalization check
+might change depending on the specializations for the underlying system.
+
+Proper deallocation of immortal instances requires distinguishing between
+statically allocated immortal instances vs those promoted by the runtime to be
+immortal. The latter which should be the only instances that require proper
+cleanup during runtime finalization.
+*/
+
+#if SIZEOF_VOID_P > 4
+/*
+In 64+ bit systems, an object will be marked as immortal by setting all of the
+lower 32 bits of the reference count field.
+
+i.e in a 64 bit system the reference count will be set to:
+    00000000 00000000 00000000 00000000
+    11111111 11111111 11111111 11111111
```

Comment:
Please place all these on the same line (otherwise it looks a bit mysterious). Perhaps also just use hex values? And/or maybe number the bits.
```suggestion
    00000000 00000000 00000000 00000000 11111111 11111111 11111111 11111111
```

Reply:
Fixed, used a hex value instead!